### PR TITLE
dbus: pass transient unit name metadata to polkit

### LINKED ISSUE #17224: In contrast to StartUnit() the StartTransientUnit() D-Bus call doesn't pass unit name metadata to PolicyKit authorization request
systemd-245.8-2.fc32.x86_64
polkit-0.116-7.fc32.x86_64

Fedora core 32 64 bit

```
systemd-run --scope -u isc_cache ccontrol start 201722B865D3
```
purpose not root user might able to start transient unit

```
cat /etc/polkit-1/rules.d/10-isc-cache.rules
polkit.addRule(function(action, subject) {
 if (action.id == "org.freedesktop.systemd1.manage-units" ) {
   polkit.log("action lookup unit=" + action.lookup("unit") );
   // !!! system-run  action.lookup("unit") undefined
   if ( action.lookup("unit") == "isc_cache.scope" && subject.isInGroup("cachemgr") ) {
     return polkit.Result.YES;
   }
  // ok systemctl start|stop isccache@<instance> for /etc/systemd/system/isccache@.service
  // purpose not root might start isc cache service if user memeber of cachemgr group
   if ( RegExp("isccache@[A-Za-z0-9_-]+.service").test(action.lookup("unit")) && subject.isInGroup("cachemgr") ) {
     return polkit.Result.YES;
   }

 }
});
```

as test code instead 
ccontrol start 201722B865D3
might be simple code

```
#include <unistd.h>
int main(int argc , char * argv [] ) {
  for(;;)
   sleep(1);
}
```
Terminate it sending SIGTERM signal.

as I understand
StartTransientUnit doesn send appropriate details when it calls CheckAuthorization.

If it is required I might add dbus-monitor --system log excerpt related to issue.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #17224.

Replaces #34598.
Closes #34598.

### CODE REVIEW

File: src/core/dbus-util.h [hunk 1]
```diff
@@ -250,7 +250,11 @@ static inline int bus_set_transient_usec(Unit *u, const char *name, usec_t *p, s
 static inline int bus_set_transient_usec_fix_0(Unit *u, const char *name, usec_t *p, sd_bus_message *message, UnitWriteFlags flags, sd_bus_error *error) {
         return bus_set_transient_usec_internal(u, name, p, true, message, flags, error);
 }
-int bus_verify_manage_units_async_full(Unit *u, const char *verb, const char *polkit_message, sd_bus_message *call, sd_bus_error *error);
+int bus_verify_manage_units_async_impl(Manager *manager, const char *id, const char *verb, const char *polkit_message, sd_bus_message *call, sd_bus_error *error);
```
Comment (YHNdnzj) [inline src/core/dbus-util.h:254]: I would probably insert a newline above

**Review by YHNdnzj — APPROVED** [review]

File: src/core/dbus-util.h [hunk 2]
```diff
@@ -250,7 +250,12 @@ static inline int bus_set_transient_usec(Unit *u, const char *name, usec_t *p, s
 static inline int bus_set_transient_usec_fix_0(Unit *u, const char *name, usec_t *p, sd_bus_message *message, UnitWriteFlags flags, sd_bus_error *error) {
         return bus_set_transient_usec_internal(u, name, p, true, message, flags, error);
 }
-int bus_verify_manage_units_async_full(Unit *u, const char *verb, const char *polkit_message, sd_bus_message *call, sd_bus_error *error);
+
+int bus_verify_manage_units_async_impl(Manager *manager, const char *id, const char *verb, const char *polkit_message, sd_bus_message *call, sd_bus_error *error);
+static inline int bus_verify_manage_units_async_full(Unit *u, const char *verb, const char *polkit_message, sd_bus_message *call, sd_bus_error *error) {
```
Comment (poettering) [inline src/core/dbus-util.h:255]: it's a bit strange that we have bus_verify_manage_units_async_full(), and bus_verify_manage_units_async_impl() here, but bus_verify_manage_units() is defined elsewhere, and entirely unrelated?

I think these really should be cleaned up if this is touched, i.e. move to the same place and be wrappers of each other?

**Comment by poettering** [comment]
lgtm, in principle, just find the split up on various .c files so weird, but i guess that can be addressed separately.

**Comment by yuwata** [comment]
https://github.com/systemd/systemd/pull/34651#discussion_r1789749195

> it's a bit strange that we have bus_verify_manage_units_async_full(), and bus_verify_manage_units_async_impl() here, but bus_verify_manage_units() is defined elsewhere, and entirely unrelated?
> 
> I think these really should be cleaned up if this is touched, i.e. move to the same place and be wrappers of each other?

Done. PTAL.

File: src/core/dbus-util.c [hunk 3]
```diff
@@ -150,30 +150,80 @@ int bus_set_transient_usec_internal(
...
                         call,
                         "org.freedesktop.systemd1.manage-units",
-                        details,
-                        &u->manager->polkit_registry,
+                        n_details > 0 ? details : NULL,
+                        &manager->polkit_registry,
+                        error);
+}
+
+int bus_verify_manage_unit_files_async(Manager *m, sd_bus_message *call, sd_bus_error *error) {
```
Comment (YHNdnzj) [inline src/core/dbus-util.c:195]: Add missing assertions while at it?

**Review by YHNdnzj — APPROVED** [review]

File: src/core/dbus-util.c (hunk 3 above)
Comment (yuwata) [reply src/core/dbus-util.c:195]: Done.

**Pull request merged**